racket/collects/racket/sequence.rkt

```diff
@@ -39,32 +41,56 @@
 (define (sequence->list s)
   (for/list ([v s]) v))
 
-(define (sequence-length s)
-  (unless (sequence? s) (raise-argument-error 'sequence-length "sequence?" s))
-  (cond [(list? s) (length s)]
+(define (sequence-length/o1 s)
```

Comment:
But `(struct? s)` will be #f for most of those, since the structs will be opaque, won't they?